### PR TITLE
[NU-1617] installation example-based e2e tests solution

### PR DESCRIPTION
**Pull request by mk-software-pl**
## Describe your changes

* real end-to-end tests solution based on the installation example docker compose
* DetectLargeTransactions & DetermineOfferedPlan scenario tests

_Explanation:_
Setup of the Nu dependencies is done by the init container. That's why all the logic is in the bash scripts. The scripts can be reused in:
* scala tests
* cypress tests
* when the solution will be moved to the Quickstart project
* manual tests

## Checklist before merge
- [ ] Related issue ID is placed at the beginning of PR title in \[brackets\] (can be GH issue or Nu Jira issue)
- [ ] Code is cleaned from temporary changes and commented out lines
- [ ] Parts of the code that are not easy to understand are documented in the code
- [ ] Changes are covered by automated tests
- [ ] Showcase in dev-application.conf added to demonstrate the feature
- [ ] Documentation added or updated
- [ ] Added entry in _Changelog.md_ describing the change from the perspective of a public distribution user
- [ ] Added _MigrationGuide.md_ entry in the appropriate subcategory if introducing a breaking change
- [ ] Verify that PR will be squashed during merge

### CODE REVIEW

File: e2e-tests/src/test/resources/spec-setup/debuggable-nu-designer.override.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+services:
+
+  designer:
+    image: touk/nussknacker:1.16.0-SNAPSHOT_scala-2.12
```
Comment (mk-software-pl) [inline e2e-tests/src/test/resources/spec-setup/debuggable-nu-designer.override.yml:4]: oh, I forgot about it. Will fix it in a minute.

**Review by lukasz-bigorajski — APPROVED** [review]

File: e2e-tests/src/test/resources/spec-setup/data/kafka/messages/transactions [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# Example messages below
```
Comment (raphaelsolarski) [inline e2e-tests/src/test/resources/spec-setup/data/kafka/messages/transactions:1]: do these files intentionally have no extension?

File: e2e-tests/src/test/scala/pl/touk/nussknacker/DetectLargeTransactionSpec.scala [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  }
+
+  override protected def afterEach(): Unit = {
+    purgeKafkaTopic("Transactions")
+    purgeKafkaTopic("ProcessedTransactions")
+    super.afterEach()
+  }
+
+  private def transactionJson(amount: Int) =
+    ujson.read(s"""{ "clientId": "100", "amount":$amount, "isLast":false }""")
```
Comment (raphaelsolarski) [inline e2e-tests/src/test/scala/pl/touk/nussknacker/DetectLargeTransactionSpec.scala:42]: `ujson.Obj("clientId" -> "100", "amount" -> amount, "isLast" -> false)`

File: e2e-tests/src/test/scala/pl/touk/nussknacker/DetermineOfferedPlanSpec.scala [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+  }
+
+  override protected def afterEach(): Unit = {
+    purgeKafkaTopic("Customers")
+    purgeKafkaTopic("SmsesWithOffer")
+    super.afterEach()
+  }
+
+  private def customerJson(name: String, age: Int, gender: String, isBigSpender: Boolean) = {
+    ujson.read(s"""
```
Comment (raphaelsolarski) [inline e2e-tests/src/test/scala/pl/touk/nussknacker/DetermineOfferedPlanSpec.scala:43]:
```scala
ujson.Obj(
  "name" -> name,
  "age" -> age,
  "gender" -> gender,
  "isBigSpender" -> isBigSpender
)
```

File: e2e-tests/src/test/scala/pl/touk/nussknacker/DetermineOfferedPlanSpec.scala [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+         |  "name": "$name",
+         |  "age": $age,
+         |  "gender": "$gender",
+         |  "isBigSpender": $isBigSpender
+         |}
+         |""".stripMargin)
+  }
+
+  private def smsWithOfferJson(name: String, offer: String) = {
+    ujson.read(s"""
```
Comment (raphaelsolarski) [inline e2e-tests/src/test/scala/pl/touk/nussknacker/DetermineOfferedPlanSpec.scala:54]:
```scala
ujson.Obj(
  "name" -> name,
  "assignedOffer" -> offer
)
```

File: e2e-tests/src/test/scala/pl/touk/nussknacker/DockerBasedInstallationExampleNuEnvironment.scala [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+  }
+
+  def sendMessageToKafka(topic: String, message: JSON): Unit = {
+    val escapedMessage = message.render().replaceAll("\"", "\\\\\"")
+    specSetupService.executeBash(s"""/app/scripts/utils/kafka/send-to-topic.sh "$topic" "$escapedMessage" """)
+  }
+
+  def readAllMessagesFromKafka(topic: String): List[JSON] = {
+    val stdout = specSetupService.executeBashAndReadStdout(s"""/app/scripts/utils/kafka/read-from-topic.sh "$topic" """)
+    stdout
```
Comment (raphaelsolarski) [inline e2e-tests/src/test/scala/pl/touk/nussknacker/DockerBasedInstallationExampleNuEnvironment.scala:47]: inline?

**Review by raphaelsolarski — APPROVED** [review]

**Pull request merged**